storage/rocksdb/ha_rocksdb.cc

```diff
@@ -2378,25 +2372,28 @@ static int rocksdb_close_connection(handlerton *const hton, THD *const thd) {
  * be used as a rocksdb transaction name
  */
 static std::string rdb_xid_to_string(const XID &src) {
-  DBUG_ASSERT(src.gtrid_length >= 0 && src.gtrid_length <= MAXGTRIDSIZE);
-  DBUG_ASSERT(src.bqual_length >= 0 && src.bqual_length <= MAXBQUALSIZE);
+  DBUG_ASSERT(src.get_gtrid_length() >= 0 &&
```

Comment:
@laurynas-biveinis Looking at xid_t in sql/xa.h, this seems like it would be reasonable to add something like an is_valid() method to the struct that validates something like``